### PR TITLE
document how to enable ipv6 for the internal network

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/all-in-one/issues/2041

### CODE REVIEW

**Comment by Zoey2936** [comment]
maybe also mention #2483? and #2476?

**Comment by szaimen** [comment]
> maybe also mention #2483? and #2476?

what do you mean?

**Comment by Zoey2936** [comment]
that the ipv6 docs also mention how to enable ipv6 in manual install / install using compose file

**Comment by Zoey2936** [comment]
not really, but maybe something like "step 3 is not needed in manual-install"

**Comment by szaimen** [comment]
I fear mentioning the manual-install will rather lead to confusion so I would not mention it for now and possibly add it later if there be posts about that coming up.

File: docker-ipv6-support.md [hunk 1]
```diff
@@ -35,6 +36,7 @@ On Windows and macOS which use Docker Desktop, you need to go into the settings,
     ```
 
 2. Add these values to the json and make sure to keep the other currently values and that you don't see `Unexpected token in JSON at position ...` before attempting to restart by clicking on `Apply & restart`.
+3. Make sure that ipv6 is enabled for the internal `nextcloud-aio` network by running `docker network inspect nextcloud-aio`. On a new instance, this command should return that it did not find a network with this name. Then you can run `docker network create --subnet="172.30.1.0/24" --subnet="2001:db8:2::/64" --driver bridge --ipv6 nextcloud-aio` in order to create the network with ipv6-support. If it finds the network though and its value `EnableIPv6` is set to false, make sure to follow https://github.com/nextcloud/all-in-one/discussions/2045 in order to recreate the network and enable ipv6 for it.
```
Comment (Zoey2936) [inline docker-ipv6-support.md:39]:
```suggestion
3. Make sure that ipv6 is enabled for the internal `nextcloud-aio` network by running `docker network inspect nextcloud-aio`. On a new instance, this command should return that it did not find a network with this name. Then you can run `docker network create --subnet="2001:db8:2::/64" --driver bridge --ipv6 nextcloud-aio` in order to create the network with ipv6-support. If it finds the network though and its value `EnableIPv6` is set to false, make sure to follow https://github.com/nextcloud/all-in-one/discussions/2045 in order to recreate the network and enable ipv6 for it.
```

File: docker-ipv6-support.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ Before you can use IPv6 in Docker containers or swarm services, you need to enab
     ```console
     sudo systemctl restart docker
     ```
+3. Make sure that ipv6 is enabled for the internal `nextcloud-aio` network by running `sudo docker network inspect nextcloud-aio`. On a new instance, this command should return that it did not find a network with this name. Then you can run `sudo docker network create --subnet="172.30.1.0/24" --subnet="2001:db8:2::/64" --driver bridge --ipv6 nextcloud-aio` in order to create the network with ipv6-support. If it finds the network though and its value `EnableIPv6` is set to false, make sure to follow https://github.com/nextcloud/all-in-one/discussions/2045 in order to recreate the network and enable ipv6 for it.
```
Comment (Zoey2936) [inline docker-ipv6-support.md:24]:
```suggestion
3. Make sure that ipv6 is enabled for the internal `nextcloud-aio` network by running `sudo docker network inspect nextcloud-aio`. On a new instance, this command should return that it did not find a network with this name. Then you can run `sudo docker network create --subnet="2001:db8:2::/64" --driver bridge --ipv6 nextcloud-aio` in order to create the network with ipv6-support. If it finds the network though and its value `EnableIPv6` is set to false, make sure to follow https://github.com/nextcloud/all-in-one/discussions/2045 in order to recreate the network and enable ipv6 for it.
```

**Review by Zoey2936 — COMMENTED** [review]
Docker can automatically select a IPv4 adress, so would it not be enough to set a IPv6 subnet(https://github.com/moby/moby/issues/41438)?

**Comment by szaimen** [comment]
> Docker can automatically select a IPv4 adress, so would it not be enough to set a IPv6 subnet([moby/moby#41438](https://github.com/moby/moby/issues/41438))?

Will it actually do so if an ipv4 subnet was not provided but only an ipv6 one?

**Comment by Zoey2936** [comment]
> > Docker can automatically select a IPv4 adress, so would it not be enough to set a IPv6 subnet([moby/moby#41438](https://github.com/moby/moby/issues/41438))?
> 
> Will it actually do so if an ipv4 subnet was not provided but only an ipv6 one?

```console
C:\Users\zoey>docker network create --subnet="2001:db8:2::/64" --driver bridge --ipv6 nextcloud-aio
9c832f9f1cb36448cdc8730dddddfd871a8f475fe8a840c55a41384c3df05c75

C:\Users\zoey>docker inspect nextcloud-aio
[
    {
        "Name": "nextcloud-aio",
        "Id": "9c832f9f1cb36448cdc8730dddddfd871a8f475fe8a840c55a41384c3df05c75",
        "Created": "2023-05-03T16:10:28.117058696Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": true,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.22.0.0/16",
                    "Gateway": "172.22.0.1"
                },
                {
                    "Subnet": "2001:db8:2::/64"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {},
        "Options": {},
        "Labels": {}
    }
]
```

**Comment by szaimen** [comment]
Okay, cool! :)

**Review by Zoey2936 — APPROVED** [review]

**Pull request merged**